src/Chapter3.hs

```diff
@@ -910,6 +1010,22 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+data Gold = Gold Int deriving (Show)
+
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (Gold g1) (Gold g2) = Gold (g1 + g2)
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append l1 l2 = l1 ++ l2
+
+instance (Append a) => Append (Maybe a) where
+    append :: Maybe a -> Maybe a -> Maybe a
+    append (Just x) (Just y) = Just (append x y)
+    append (Just x) Nothing  = Just x
+    append Nothing (Just y)  = Just y
+    append Nothing  Nothing  = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂

Reply:
Ah! I have not noticed it. I updated the code based on your suggestion.